cpp/include/cudf/datetime.hpp

```diff
@@ -469,5 +469,96 @@ std::unique_ptr<column> floor_nanosecond(
   column_view const& column,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Round down to the nearest day
```

Comment:
This doesn't necessarily round down. (applies below as well)
```suggestion
 * @brief Round to the nearest day
```

Reply:
Thanks Bradley for the reviews! I think I get your vision: we want to introduce the same abstraction we have now in our C++ layer, in the Cython layer, sort of like a dispatcher. Am I understanding this correctly? we can certainly chat more about this offline :)